src/modules/admin/components/AdminDashboard/AdminDashboard.tsx

```diff
@@ -172,15 +193,20 @@ const AdminDashboard = ({
     return <LoadingTemplate loadingText={MSG.loadingText} />;
   }
 
-  if (!canAdminister(rootRoles)) {
+  if (!hasRoot(rootUserRoles) && !canArchitect(allUserRoles)) {
     return <Redirect to={CURRENT_COLONY_ROUTE} />;
   }
 
   const { displayName } = colony;
   return (
     <div className={styles.main}>
       <VerticalNavigation
-        navigationItems={navigationItems(colony, domains, rootRoles)}
+        navigationItems={navigationItems(
+          colony,
+          domains,
+          rootUserRoles,
+          allUserRoles,
+        )}
```

Comment:
Would it make sense to `useMemo` with this fn call?

Reply:
I don't think so. It doesn't do a very heavy calculation. Just some conditions, that should be very fine.